weave-js/src/components/PagePanelComponents/Home/Browse3/compare/ShoppingCart.tsx

```diff
@@ -0,0 +1,150 @@
+/**
+ * This name is an analogy - the collection of items that you have
+ * "put in your basket" for comparison.
+ */
+
+import React from 'react';
+import {useHistory} from 'react-router-dom';
+import {SortableContainer} from 'react-sortable-hoc';
+
+import {queryToggleString, searchParamsSetArray} from '../urlQueryUtil';
+import {ShoppingCartItem} from './ShoppingCartItem';
+import {ShoppingCartItemDefs} from './types';
+
+type ShoppingCartItemsProps = {
+  items: ShoppingCartItemDefs;
+  baselineEnabled: boolean;
+  selected: string | null;
+};
+
+type SortableItemsProps = ShoppingCartItemsProps & {
+  onClickShoppingCartItem: (value: string) => void;
+  onSetBaseline: (value: string | null) => void;
+  onRemoveShoppingCartItem: (value: string) => void;
+};
+
+const SortableItems = SortableContainer(
+  ({
+    items,
+    baselineEnabled,
+    selected,
+    onClickShoppingCartItem,
+    onSetBaseline,
+    onRemoveShoppingCartItem,
+  }: SortableItemsProps) => {
+    return (
+      <div className="flex flex-wrap items-center gap-8">
+        {items.map((item, index) => (
+          <ShoppingCartItem
+            key={`item-${item.value}`}
+            numItems={items.length}
+            index={index}
+            idx={index}
+            item={item}
+            baselineEnabled={baselineEnabled}
+            onClickShoppingCartItem={onClickShoppingCartItem}
+            onSetBaseline={onSetBaseline}
+            onRemoveShoppingCartItem={onRemoveShoppingCartItem}
+            isSelected={item.value === selected}
+          />
+        ))}
+      </div>
+    );
+  }
+);
+
+// Create a copy of the specified array, moving an item from one index to another.
+function arrayMove<T>(array: readonly T[], from: number, to: number) {
+  const slicedArray = array.slice();
+  slicedArray.splice(
+    to < 0 ? array.length + to : to,
+    0,
+    slicedArray.splice(from, 1)[0]
+  );
+  return slicedArray;
+}
+
+export const ShoppingCart = ({
+  items,
+  baselineEnabled,
+  selected,
+}: ShoppingCartItemsProps) => {
+  const history = useHistory();
+
+  const onSortEnd = ({
+    oldIndex,
+    newIndex,
+  }: {
+    oldIndex: number;
+    newIndex: number;
+  }) => {
+    if (oldIndex === newIndex) {
+      return;
+    }
+    const {search} = history.location;
```

Comment:
I noticed this elsewhere, but figured I would add one comment: I generally stay way from embedding URL management deep in components like this. Ideally the URL management would happen in a very high level top component (Browse3) and everything below is driven by params and callbacks. The reason for this is that if we ever want to export to reports or embed in other contexts where the state is no longer URL managed, then we incur a pretty hefty refactor.